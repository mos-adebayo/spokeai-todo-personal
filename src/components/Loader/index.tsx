import React from 'react';
import {useSelector} from "react-redux";
import {RootState} from "../../redux/reducers/rootReducers";
import {Spinner} from "react-bootstrap";
import {Wrapper} from "./styles";

const Index: React.FC = () => {
    const { loading } = useSelector((state: RootState) => state.tasks)

    if(!loading) return <React.Fragment />

    return (
        <Wrapper>
            <Spinner animation="border" role="status">
                <span className="visually-hidden">Loading...</span>
            </Spinner>
        </Wrapper>
    );
};

export default Index;
