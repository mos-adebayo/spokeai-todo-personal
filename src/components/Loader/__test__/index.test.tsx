import React from "react";
import { shallow } from "enzyme";
import Loader from "../index";

describe("<Loader />", () => {
  it("should match snapshot when loading is true", () => {
    const wrapper = shallow(<Loader isLoading />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot when loading is false", () => {
    const wrapper = shallow(<Loader isLoading={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
