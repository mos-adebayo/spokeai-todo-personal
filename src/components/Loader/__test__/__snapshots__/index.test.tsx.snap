// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match snapshot when loading is false 1`] = `<Fragment />`;

exports[`<Loader /> should match snapshot when loading is true 1`] = `
<styled.div>
  <Spinner
    animation="border"
    role="status"
  >
    <span
      className="visually-hidden"
    >
      Loading...
    </span>
  </Spinner>
</styled.div>
`;
